python/plugins/processing/gui/AlgorithmExecutor.py

```diff
@@ -178,7 +183,7 @@ def execute_in_place_run(alg, parameters, context=None, feedback=None, raise_exc
                 else:
                     active_layer.deleteFeature(f.id())
                     # Get the new ids
-                    old_ids = set([f.id() for f in active_layer.getFeatures(req)])
+                    old_ids = set([f.id() for f in active_layer.getFeatures()])
```

Comment:
This change looks potentially very expensive -- can it be tightened at all?

Reply:
Yes: I wanted to add 'req' back but ran out of time yesterday night. I will fix it later today.